libs/@guardian/source-foundations/src/utils/convert-value.ts

```diff
@@ -1,5 +1,9 @@
 export const rootPixelFontSize = 16;
+
 export const pxToRem = (px: number): number => px / rootPixelFontSize;
-export const pxStringToNumber = (px: string): number => Number(px.slice(0, -2));
+
+export const pxStringToNumber = <N extends number>(px: `${N}px`): N =>
+	Number(px.slice(0, -2)) as N;
```

Comment:
I think that whenever we us `as` assertions we should complement it with a series of unit tests that guarantee it will actually return this value!

Reply:
Agreed. There's already a unit test, and to be honest I can't think of any other cases that I could add that respect the input type `${number}px`...

Reply:
Just leaving this here for the enthusiasts: https://github.com/dubzzz/fast-check